Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php

```diff
@@ -302,4 +333,68 @@ protected function renderTraitsCode()
 
         return '    use ' . implode(', ', $this->traits) . ";\n\n";
     }
+
+    /**
+     * @param $proxyClassName
+     * @param string $originalClassName
+     * @return string
+     * @throws ReflectionException
+     */
+    protected function buildLazyProxyClass($proxyClassName, string $originalClassName): string
+    {
+        $methods = [$this->buildCallMagicMethod()];
+
+        foreach (get_class_methods($this->fullOriginalClassName) as $methodName) {
+            if (!$this->reflectionService->isMethodPublic($this->fullOriginalClassName, $methodName) || $methodName === '__call' || $methodName === '__construct') {
+                continue;
+            }
+            $methodReturnType = $this->reflectionService->getMethodDeclaredReturnType($this->fullOriginalClassName, $methodName);
+            $returnKeyword = ($methodReturnType === 'void') ? '' : 'return ';
+            $method = MethodGenerator::fromReflection(new \Laminas\Code\Reflection\MethodReflection($this->fullOriginalClassName, $methodName));
+            $method->removeDocBlock();
+            $method->setBody(
+                <<< CODE
+                    \$arguments = func_get_args();
+                    {$returnKeyword}\$this->_activateDependency()->{$methodName}(...\$arguments);
+                CODE
+            );
+            $methods[] = $method;
+        }
+
+        $classGenerator = ClassGenerator::fromArray(['name' => $proxyClassName . '_LazyProxy'])
+            ->setExtendedClass($proxyClassName)
+            ->setImplementedInterfaces([DependencyProxy::class])
+            ->removeMethod('__call')
+            ->addMethods($methods)
+            ->addTrait('\\' . DependencyProxyTrait::class);
+        // This is just for making Psalm happy, since addTrait() returns TraitUsageInterface which doesn't provide generate()
+        assert($classGenerator instanceof ClassGenerator);
```

Comment:
isnt this `assert` odd? - does php doc suffice? (im just curious - shouldnt stop this pr ^^)
```suggestion
        /** @var $classGenerator ClassGenerator&TraitUsageInterface */
```

Reply:
Should work as well to make psalm understand, yes. The assert is more bullet-proof though (actual runtime check) and is optimized away in production anyway.